react/week4/my-app/src/users.js

```diff
@@ -0,0 +1,33 @@
+import React from "react";
+import  {UserContext} from "./user.context";
+
+ const Users = () => {
+  
+    const {users, loading, error} = React.useContext(UserContext);
+  
+    if (error) {
+      return <div>{`error fetching ${error}...`}</div>;
+    }
+  
+    if (loading) {
+      return <div>Loading...</div>;
+    }
+  
+    if (!users || users.length === 0) {
```

Comment:
Is there any situation where `!users` is needed? I think that users is always a list. If it is not, we should probably try to validate the result of the API call to make it so.

In general, we strive to keep the state (in a component, context or redux) of the same type throughout the application.

Reply:
Actually before i added (!users) when i typing in the input, it all works good, but when i delete the content of the input i got this error (TypeError: Cannot read property 'length' of undefined).

Reply:
I imagined that there was a good reason for it to be there :)

Ideally, I would track it down fix it (wherever it is being assigned no `undefined`). It is fine that you do not do it for the homework, but keep the idea in mind: strive to keep the state of the same type throughout the lifetime of the application. Only use `undefined` or `null` when it represents a specific situation (e.g. user id for an anonymous user is often `undefined` or `null).